.github/actions/cargo_target_dir_pre_cache/action.yml

```diff
@@ -0,0 +1,13 @@
+name: CARGO_TARGET_DIR Pre Cache
+description: CARGO_TARGET_DIR Pre Cache
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: |
+        crates=($(cargo metadata --format-version 1 | jq -r '.workspace_members[] | split(" ")  | .[0] | gsub("-";"_")'))
```

Comment:
nit: you don't need deps for this `cargo metadata --no-deps --format-version 1` should finish much faster, since it doesn't need to process our deps (and sometimes perform a subsequent crates.io fetch)

Reply:
oh cool, will update shortly.